specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/Assessments/PutAssessment_example.json

```diff
@@ -0,0 +1,37 @@
+{
+  "parameters": {
+    "api-version": "2019-01-01-preview",
+    "resourceId": "subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourceGroups/myRg/providers/Microsoft.Compute/virtualMachineScaleSets/vmss2",
+    "assessmentName": "8bb8be0a-6010-4789-812f-e4d661c4ed0e",
+    "body": {
+      "properties": {
+        "resourceDetails": {
+          "source": "Azure",
```

Comment:
Making the user provide "source" and "id" when it can be inferred from the URI the assessment is created on seems unnecessarily painful.

Reply:
this is true only to Azure
we have to get these details when the resource is not Azure and we didn't want to create different behavior
WDYT?

Reply:
You have different models for the different "source" values.  Just make "id" either readonly or not required in the AzureResourceDetails and populate it from the URI.

Reply:
done :)